src/components/pages/UserBooksPage/UserBooks.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from "react";
+import { FC, useState, useEffect, useContext } from "react";
+import Book_list from "../../../interfaces/book_list.interface";
+import Book from "../../../interfaces/book.interface";
+import Book_list_entry from "../../../interfaces/book_list_entry.interface";
+import { useNavigate, useParams } from "react-router-dom";
+import { TheContext } from "../../../TheContext";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import {
+    Paper,
+    Typography,
+    Button,
+    Stack,
+    Fab,
+    Card,
+    CardActionArea
+} from "@mui/material";
+import {
+    fetchBook,
+    fetchAllBooks,
+    fetchAllEntries,
+    fetchDeleteEntry,
+    fetchEntry,
+    fetchEntriesByList
+} from "../../../fetchFunctions";
+import { listBooksDeleteButton, userPageBackButton } from "../../../sxStyles";
+
+const UserBooks: FC<{ booklist: Book_list, handleCloseList: Function }> = ({ booklist, handleCloseList }): JSX.Element => {
+    const [books, setBooks] = useState<Book[]>([]);
+    const [entries, setEntries] = useState<Book_list_entry[]>([]);
+    const context = useContext(TheContext);
+    const navigate = useNavigate();
+
+    const fetchBooks = async () => setBooks(await fetchAllBooks());
+    const fetchEntries = async () => setEntries(await fetchEntriesByList(booklist.id));
+
+    useEffect(() => {
+        if (booklist.id !== -1){   
+            fetchBooks();
+            fetchEntries();
+        }
+    }, [booklist]);
+
+    const renderEntriesInList = () => {
+        let renderedBooks = [];
+        for (const entry of entries) {
+            for (const book of books) {
```

Comment:
Would make more sense to order books into a keyvalue array and then just get them by id instead of looping through all books n times